nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -372,10 +372,7 @@ in
       preStart =  mkOption {
         type = types.lines;
         default = ''
-          test -d ${cfg.stateDir}/logs || mkdir -m 750 -p ${cfg.stateDir}/logs  
-          test `stat -c %a ${cfg.stateDir}` = "750" || chmod 750 ${cfg.stateDir}
-          test `stat -c %a ${cfg.stateDir}/logs` = "750" || chmod 750 ${cfg.stateDir}/logs
-          chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
+          ${cfg.package}/bin/nginx -c '${configFile}' -p '${cfg.stateDir}' -t
```

Comment:
Maybe add `defaultText` to not have the hardcoded nginx package path

Reply:
Fixed config. 
This variant not worked
```
      preStart =  mkOption {
        type = types.lines;
        default = ''
          ${cfg.package}/bin/nginx -c '${configFile}' -p '${cfg.stateDir}' -t
        '';

```